frontend/src/components/Button/CopyButton.js

```diff
@@ -5,13 +5,29 @@ import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faCopy } from '@fortawesome/free-solid-svg-icons';
 import { CopyToClipboard } from 'react-copy-to-clipboard';
 import { Button as CarbonButton } from 'carbon-components-react';
+import { connect } from 'react-redux';
+import { setInlineNotification } from '../../actions/exerciseActions';
 
-const CopyButton = ({ copyText }: { copyText: string }) => (
-  <CarbonButton kind="secondary">
-    <CopyToClipboard text={copyText}>
+const CopyButton = ({
+  copyText,
+  setInlineNotification,
+}: {
+  copyText: string,
+  setInlineNotification: (inlineNotification: string) => void,
+}) => (
+  <CopyToClipboard text={copyText}>
+    <CarbonButton
+      kind="secondary"
+      onClick={() => setInlineNotification && setInlineNotification('copied')}
+    >
       <FontAwesomeIcon icon={faCopy} />
-    </CopyToClipboard>
-  </CarbonButton>
+    </CarbonButton>
+  </CopyToClipboard>
 );
 
-export default CopyButton;
+export default connect(
+  state => ({}),
```

Comment:
this can just be `null`
```suggestion
  null,
```